shell/platform/android/test/io/flutter/embedding/android/FlutterActivityAndFragmentDelegateTest.java

```diff
@@ -0,0 +1,502 @@
+package io.flutter.embedding.android;
+
+import android.app.Activity;
+import android.arch.lifecycle.Lifecycle;
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.Robolectric;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.FlutterShellArgs;
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.embedding.engine.plugins.activity.ActivityControlSurface;
+import io.flutter.embedding.engine.renderer.FlutterRenderer;
+import io.flutter.embedding.engine.systemchannels.AccessibilityChannel;
+import io.flutter.embedding.engine.systemchannels.LifecycleChannel;
+import io.flutter.embedding.engine.systemchannels.LocalizationChannel;
+import io.flutter.embedding.engine.systemchannels.NavigationChannel;
+import io.flutter.embedding.engine.systemchannels.SettingsChannel;
+import io.flutter.embedding.engine.systemchannels.SystemChannel;
+import io.flutter.plugin.platform.PlatformPlugin;
+import io.flutter.plugin.platform.PlatformViewsController;
+import io.flutter.view.FlutterMain;
+
+import static android.content.ComponentCallbacks2.TRIM_MEMORY_RUNNING_LOW;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@Config(manifest=Config.NONE)
+@RunWith(RobolectricTestRunner.class)
+public class FlutterActivityAndFragmentDelegateTest {
+  private FlutterEngine mockFlutterEngine;
+  private FakeHost fakeHost;
+  private FakeHost spyHost;
+
+  @Before
+  public void setup() {
+    // FlutterMain is utilized statically, therefore we need to inform it to behave differently
+    // for testing purposes.
+    FlutterMain.setIsRunningInRobolectricTest(true);
+
+    // Create a mocked FlutterEngine for the various interactions required by the delegate
+    // being tested.
+    mockFlutterEngine = mockFlutterEngine();
+
+    // Create a fake Host, which is required by the delegate being tested.
+    fakeHost = new FakeHost();
+    fakeHost.flutterEngine = mockFlutterEngine;
+
+    // Create a spy around the FakeHost so that we can verify method invocations.
+    spyHost = spy(fakeHost);
```

Comment:
otherwise, you also end up having to do this

Reply:
we can update it in the future if a mock seems like a better direction. For now it's probably better that we make some forward progress.